### PR TITLE
add client code

### PR DESCRIPTION
**Pull request by rogpeppe**
This has all the client code except the client tests, which are moved to the server package
as they need the right server to run against.

### CODE REVIEW

File: LICENCE [hunk 1]
```diff
@@ -0,0 +1,661 @@
+                    GNU AFFERO GENERAL PUBLIC LICENSE
```
Comment (cmars) [inline LICENCE]: @urosj LGPL or AGPL?

**Comment by mattyw** [comment]
LGTM providing @urosj confirms what license we want

**Comment by urosj** [comment]
:+1:

File: idmtest/idmtest.go [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+	}
+	router := httprouter.New()
+	for _, route := range errorMapper.Handlers(func(httprequest.Params) (*handler, error) {
+		return h, nil
+	}) {
+		router.Handle(route.Method, route.Path, route.Handle)
+	}
+	mux := http.NewServeMux()
+	httpbakery.AddDischargeHandler(mux, "/v1/discharger", srv.bakery, srv.check)
+	router.Handler("POST", "/v1/discharger/*rest", mux)
```
Comment (mhilton) [inline idmtest/idmtest.go]: There is an extension for this to allow for the /publickey and /discharge endpoints. We need to make sure it ends up here (not in this PR though).

**Comment by mhilton** [comment]
:+1:

**Pull request merged**